types/event.d.ts

```diff
@@ -266,7 +271,7 @@ export interface EventMap {
       currentGP: number;
       maxGP: number;
       currentShield: number;
-      jobDetail: JobDetail[T];
+      jobDetail: JobDetail['PLD'] & JobDetail['WAR'] & JobDetail['DRK'] & JobDetail['GNB'] & JobDetail['WHM'] & JobDetail['SCH'] & JobDetail['AST'] & JobDetail['PGL'] & JobDetail['MNK'] & JobDetail['DRG'] & JobDetail['NIN'] & JobDetail['SAM'] & JobDetail['BRD'] & JobDetail['MCH'] & JobDetail['DNC'] & JobDetail['THM'] & JobDetail['BLM'] & JobDetail['ACN'] & JobDetail['SMN'] & JobDetail['RDM'];
```

Comment:
Is it possible to just call this `JobDetail<Job>`?

Also, I think it might make sense to remove the `T` parameterization and just have `job: Job` up above.